server.go

```diff
@@ -243,7 +244,16 @@ func (s *Server) NewMiddlewareHandler(middlewares []Middleware) http.Handler {
 					w: res,
 				},
 				RequestID: func() string {
-					return requestID
+					idRaw := requestCtx[RequestIDKey]
+					if id, ok := idRaw.(string); ok {
+						return id
+					} else {
+						// ID not found in map
+						return ""
+					}
+				},
+				SetRequestID: func(ID string) {
```

Comment:
What about defining the getter/setter on the Context type? That way the handling of internal information would be nicer.